### PR TITLE
Add new features, fix .env setup, fix permissions and UID conflicts

### PR DESCRIPTION
**Pull request by Pavlogal**
I've addressed all complaints from https://github.com/Luctia/ezarr/pull/49 and some more.

Changes:
- Re-added support for Jackett, thus addressing: https://github.com/Luctia/ezarr/pull/10#issuecomment-1442821059
- Added FlareSolverr, thus solving https://github.com/Luctia/ezarr/issues/48#issue-2213261081
- Fixed audiobookshelf, thus solving https://github.com/Luctia/ezarr/issues/44#issue-2129818996
- Added automatic timezone detection feature to `main.py` (it just reads from /etc/timezone),
- The python script can now act as a compose file generator, as it asks the user if they actually want to set up folders and permissions.
- **Resolved UID conflicts, while keeping the others the same as before**
- **Fixed manual setup by making new `.env` and `docker-compose.yml` samples. This also gives the end user the ability to customize container UIDs for manual installs.** I also changed the provided compose file to a `.sample`, so it's more like the .env sample and the user will always have a copy of the original even if overwritten by the python script.
- Since we're endorsing compose v2 I removed `version` from `docker-compose.yml` so compose v2 doesn't complain about it. Also added a note to README.
- Added a script to remove old users and an update script
- Updated README to reflect changes and added a note for NFS users to save them some trouble

I tried to keep my commit history as clean as possible so I hope reviewing this will be easy, although I did make quite a lot of changes. I did some testing and everything should work perfectly now.

### CODE REVIEW

**Comment by Pavlogal** [comment]
It failed the docker-compose check because there is no longer a docker-compose file but just a sample. It wouldn't work anyway now since `docker-compose.yml` now depends on `.env` for manual installs. Checks should be rewritten or I can simply add a python-generated compose file which is hardcoded and should work in the current workflow.

**Comment by Ulimn** [comment]
> It failed the docker-compose check because there is no longer a docker-compose file but just a sample. It wouldn't work anyway now since `docker-compose.yml` now depends on `.env` for manual installs. Checks should be rewritten or I can simply add a python-generated compose file which is hardcoded and should work in the current workflow.

You could add something like a "sample" `ci-compose.yml` and use it in the CI with the `-f` flag.

File: setup.sh [hunk 1]
```diff
@@ -1,20 +1,38 @@
 #!/bin/bash
 
+# Set up your .env file BEFORE running this script!!!
+# export all variables from .env
+set -a
+source .env
+set +a
```
Comment (Luctia) [inline setup.sh:7]: I'm confused what this does, could you explain it? I think [this](https://stackoverflow.com/a/30969768/7443523) is what you're doing, but it uses `-o`.

File: setup.sh [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
+sudo useradd sabnzbd -u $SABNZBD_UID
+sudo useradd jellyseerr -u $SABNZBD_UID
+sudo useradd bazarr -u $BAZARR_UID
+sudo useradd audiobookshelf -u $AUDIOBOOKSHELF_UID
+sudo groupadd mediacenter -g $MEDIACENTER_GID
+
+# Adds current user to the mediacenter group. This is recommended so that you can still have access to files inside the ezarr folder structure for manual control.
+# This is way better than just doing everything as root, especially on NFS shares. Also some services run as the default user anyway (Jellyfin, Tautulli).
+sudo usermod -a -G mediacenter $USER
+# When you add the user to the group the cahanges don't take effect immediately.
```
Comment (Luctia) [inline setup.sh:32]: Nit: typo

File: setup.sh [hunk 3]
```diff
@@ -46,8 +68,11 @@ sudo chown -R qbittorrent:mediacenter config/qbittorrent-config
...
 sudo chown -R plex:mediacenter config/plex-config
+sudo chown -R $UID:mediacenter config/jellyfin-config
+sudo chown -R $UID:mediacenter config/tautulli-config
 sudo chown -R sabnzbd:mediacenter config/sabnzbd-config
 sudo chown -R jellyseerr:mediacenter config/jellyseerr-config
 sudo chown -R bazarr:mediacenter config/bazarr-config
+sudo chown -R audiobookshelf:mediacenter config/audiobookshelf-config
 
-echo "UID=$(id -u)" >> .env
+echo "Done! It is recommended to reboot now."
```
Comment (Luctia) [inline setup.sh:78]: Nit: end file with empty line

File: users_groups_setup.py [hunk 4]
```diff
@@ -66,20 +64,16 @@ def mylar3(self):
...
     def audiobookshelf(self):
         os.system(
-            '/bin/bash -c "sudo useradd audiobookshelf -u 13009'
-            ' ; sudo mkdir -pv ' + self.root_dir + '/data/{media,usenet,torrents}/audiobooks -m 775'
-            ' ; sudo chown -R audiobookshelf:mediacenter ' + self.root_dir + '/data/{media,usenet,torrents}/audiobooks'
-            ' ; sudo chown $(id -u):mediacenter ' + self.root_dir + '/data'
-            ' ; sudo chown $(id -u):mediacenter ' + self.root_dir + '/data/{media,usenet,torrents}"'
+            '/bin/bash -c "sudo useradd audiobookshelf -u 13014'
+            ' ; sudo mkdir -pv ' + self.root_dir + '/data/media/{audiobooks,podcasts,audiobookshelf-metadata} -m 775'
+            ' ; sudo chown -R audiobookshelf:mediacenter ' + self.root_dir + '/data/media/{audiobooks,podcasts,audiobookshelf-metadata}"'
```
Comment (Luctia) [inline users_groups_setup.py:76]: Do we also want to put stuff here for torrents/usenet? I'm not sure whether audiobookshelf has a feature to move stuff when torrents/usenet downloads (?) complete like Sonarr/Radarr does, but we do make those for movies and tv shows.

**Review by Luctia — CHANGES_REQUESTED** [review]
Sorry for taking so long! Life got in the way. Thanks again for your effort, I hope my comments don't demotivate you, they're all pretty small :)

FYI: comments marked by "nit" are nitpicks that you can ignore, but would probably be nice if changed.

Just out of curiosity: did you test the new scripts (`remove_old_users.sh` and the new `setup.sh`)?

**Comment by Ulimn** [comment]
Just a small note: I actually used the `remove_old_users.sh` script to clean up the mess left by the original that failed. It worked for me.

**Comment by Luctia** [comment]
@Pavlogal I really like your work apart from my comments, and would like to merge it. Could you let me know when you think you have time to work on the changes? If I don't hear anything before the end of the week, I think I'll merge it anyway and just immediately apply the changes myself.

**Comment by Pavlogal** [comment]
Sorry for inactivity, I'm in the middle of college exam period and will have a little bit of time on the 21st/22nd of june. Let me know if that's okay for you

**Comment by Luctia** [comment]
Absolutely! No need to apologize, I was just curious. That's absolutely fine, good luck on your exams!

**Comment by Luctia** [comment]
@Pavlogal do you think you'll get around to it soon, or should I pick it up?

**Comment by Pavlogal** [comment]
I'll start work on it later today

**Comment by Pavlogal** [comment]
Sorry for the delays, college can be a bitch... I've addressed all the nitpicks. As for audiobookshelf, iirc I was looking into their docs and I couldn't find any mention of `AUDIOBOOKSHELF_UID` and `AUDIOBOOKSHELF_GID` like it was in your config. It really annoyed me how poorly documented this was. Their official examples actually always use either a `user` environment variable in the format UID:GID or a user section in docker compose with the same format:
https://github.com/advplyr/audiobookshelf/blob/master/docker-compose.yml
https://www.audiobookshelf.org/guides/docker-install/
I chose the first option and it seems to work.

I copied the folder structure directly from their docs so it has room for everything it needs, moreover audiobookshelf doesn't have the feature to look for, download and hardlink audiobooks. It's more like a book management service.

I did test the new user scripts back when I opened this PR and they worked really well. You just need to reboot sometimes for changes to fully take effect.

**Review by Luctia — APPROVED** [review]

**Comment by Luctia** [comment]
Big thanks to @Pavlogal for these changes! As far as I can see, there's no more issues and it solves quite a few inconsistencies.

**Pull request merged**